smartsim/_core/launcher/step/step.py

```diff
@@ -81,15 +83,11 @@ def get_colocated_launch_script(self) -> str:
         else:
             db_log_file = "/dev/null"
 
-        # if user specified to use taskset with local launcher
-        # (not allowed b/c MacOS doesn't support it)
-        # TODO: support this only on linux
-        if (
-            self.__class__.__name__ == "LocalStep"
-            and db_settings["limit_app_cpus"] is True
-        ):  # pragma: no cover
-            logger.warning("Setting limit_app_cpus=False for local launcher")
-            db_settings["limit_app_cpus"] = False
+        if sys.platform == 'darwin':
+            if isinstance(self, LocalStep) and db_settings["limit_db_cpus"] is True:
```

Comment:
style thing: If you want to save a layer of nesting:
```py
if sys.platform == "darwin" and isinstance(self, LocalStep) and ...:
```

Reply:
follow up question: should we even bother with the downcast to local step? If a user is on macos, I'm pretty sure we are safe to assume that they are using a local launcher (or if they aren't, they're likely to have much bigger issues, hahaha)

Reply:
Good point. It does seem odd to check whether a step is a LocalStep in just the colocated part of the code.